controllers/druid/interface.go

```diff
@@ -0,0 +1,172 @@
+package druid
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/druid-io/druid-operator/apis/druid/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	resourceCreated = "CREATED"
+	resourceUpdated = "UPDATED"
+)
+
+const (
+	DruidNodeUpdateFail            string = "UPDATE_FAIL"
+	DruidNodeUpdateSuccess         string = "UPDATE_SUCCESS"
+	DruidNodeRollingDeploymentWait string = "ROLLING_DEPLOYMENT_WAIT"
+	DruidNodeDeleteFail            string = "DELETE_FAIL"
+	DruidNodeDeleteSuccess         string = "SUCCESS_FAIL"
+	DruidNodeCreateSuccess         string = "CREATE_SUCCESS"
+	DruidNodeCreateFail            string = "CREATE_FAIL"
+	DruidNodePatchFail             string = "PATCH_FAIL"
+	DruidSpecInvalid               string = "SPEC_INVALID"
+	DruidNodeRunning               string = "RUNNING"
+	DruidObjectListFail            string = "LIST_FAIL"
+	DruidOjectGetFail              string = "GET_FAIL"
+	DruidFinalizerSuccess          string = "TRIGGER_FINALIZER_SUCCESS"
+	DruidFinalizer                 string = "TRIGGER_FINALIZER"
+)
+
+// Reader Interface
+type Reader interface {
+	List(ctx context.Context, sdk client.Client, drd *v1alpha1.Druid, selectorLabels map[string]string, emptyListObjFn func() runtime.Object, ListObjFn func(obj runtime.Object) []object) ([]object, error)
+	Get(ctx context.Context, sdk client.Client, nodeSpecUniqueStr string, drd *v1alpha1.Druid, emptyObjFn func() object) (object, error)
+}
+
+// Writer Interface
+type Writer interface {
+	Delete(ctx context.Context, sdk client.Client, drd *v1alpha1.Druid, obj runtime.Object, deleteOptions ...client.DeleteOption) error
+	Create(ctx context.Context, sdk client.Client, drd *v1alpha1.Druid, obj object) (string, error)
+	Update(ctx context.Context, sdk client.Client, drd *v1alpha1.Druid, obj object) (string, error)
+	Patch(ctx context.Context, sdk client.Client, drd *v1alpha1.Druid, obj object, status bool, patch client.Patch) error
+}
+
+// WriterFuncs struct
+type WriterFuncs struct {
+	deleteFunc func(ctx context.Context, sdk client.Client, drd *v1alpha1.Druid, obj runtime.Object) error
+	createFunc func(ctx context.Context, sdk client.Client, drd *v1alpha1.Druid, obj object) (string, error)
+	updateFunc func(ctx context.Context, sdk client.Client, drd *v1alpha1.Druid, obj object) (string, error)
+	patchFunc  func(ctx context.Context, sdk client.Client, drd *v1alpha1.Druid, obj object, status bool, patch client.Patch) error
+}
+
+// ReaderFuncs struct
+type ReaderFuncs struct {
+	listFunc func(ctx context.Context, sdk client.Client, drd *v1alpha1.Druid, selectorLabels map[string]string, emptyListObjFn func() runtime.Object, ListObjFn func(obj runtime.Object) []object) ([]object, error)
+	getFunc  func(ctx context.Context, sdk client.Client, nodeSpecUniqueStr string, drd *v1alpha1.Druid, emptyObjFn func() object) (object, error)
```

Comment:
The same here, I see they aren't assigned anywhere.

Reply:
agreed